src/overlays/gamestates/ovl_file_choose/file_select.h

```diff
@@ -0,0 +1,335 @@
+    /* 3 */ FS_TITLE_COPY_TO,       // "Copy to which file?"
+    /* 4 */ FS_TITLE_COPY_CONFIRM,  // "Are you sure?"
+    /* 5 */ FS_TITLE_COPY_COMPLETE, // "File copied."
+    /* 6 */ FS_TITLE_ERASE_FILE,    // "Erase which file?"
+    /* 7 */ FS_TITLE_ERASE_CONFIRM, // "Are you sure?"
+    /* 8 */ FS_TITLE_ERASE_COMPLETE // "File erased."
+} TitleLabel;
+
+typedef enum {
+    /* -1 */ FS_WARNING_NONE = -1,
+    /*  0 */ FS_WARNING_NO_FILE_COPY,   // "No file to copy."
+    /*  1 */ FS_WARNING_NO_FILE_ERASE,  // "No file to erase."
+    /*  2 */ FS_WARNING_NO_EMPTY_FILES, // "There is no empty file."
+    /*  3 */ FS_WARNING_FILE_EMPTY,     // "This is an empty file."
+    /*  4 */ FS_WARNING_FILE_IN_USE     // "This file is in use."
+} WarningLabel;
+
+typedef enum {
+    /* 0 */ FS_BTN_MAIN_FILE_1,
+    /* 1 */ FS_BTN_MAIN_FILE_2,
+    /* 2 */ FS_BTN_MAIN_FILE_3,
+    /* 3 */ FS_BTN_MAIN_COPY,
+    /* 4 */ FS_BTN_MAIN_ERASE,
+    /* 5 */ FS_BTN_MAIN_OPTIONS
+} MainMenuButtonIndex;
+
+typedef enum {
+    /* 0 */ FS_BTN_COPY_FILE_1,
+    /* 1 */ FS_BTN_COPY_FILE_2,
+    /* 2 */ FS_BTN_COPY_FILE_3,
+    /* 3 */ FS_BTN_COPY_QUIT
+} CopyMenuButtonIndex;
+
+typedef enum {
+    /* 0 */ FS_BTN_ERASE_FILE_1,
+    /* 1 */ FS_BTN_ERASE_FILE_2,
+    /* 2 */ FS_BTN_ERASE_FILE_3,
+    /* 3 */ FS_BTN_ERASE_QUIT
+} EraseMenuButtonIndex;
+
+typedef enum {
+    /* 0 */ FS_BTN_SELECT_FILE_1,
+    /* 1 */ FS_BTN_SELECT_FILE_2,
+    /* 2 */ FS_BTN_SELECT_FILE_3,
+    /* 3 */ FS_BTN_SELECT_YES,
+    /* 4 */ FS_BTN_SELECT_QUIT
+} SelectMenuButtonIndex;
+
+typedef enum {
+    /* 0 */ FS_BTN_CONFIRM_YES,
+    /* 1 */ FS_BTN_CONFIRM_QUIT
+} ConfirmButtonIndex;
+
+typedef enum {
+    /* 0 */ FS_BTN_ACTION_COPY,
+    /* 1 */ FS_BTN_ACTION_ERASE
+} ActionButtonIndex;
+
+typedef enum {
+    /* 0 */ FS_SETTING_AUDIO,
+    /* 1 */ FS_SETTING_ZTARGET
+} SettingIndex;
+
+typedef enum {
+    /* 0 */ FS_CHAR_PAGE_HIRA,
+    /* 1 */ FS_CHAR_PAGE_KATA,
+    /* 2 */ FS_CHAR_PAGE_ENG
+} CharPage;
+
+typedef enum {
+    /*  0 */ FS_KBD_BTN_HIRA,
+    /*  1 */ FS_KBD_BTN_KATA,
+    /*  2 */ FS_KBD_BTN_ENG,
+    /*  3 */ FS_KBD_BTN_BACKSPACE,
+    /*  4 */ FS_KBD_BTN_END,
+    /* 99 */ FS_KBD_BTN_NONE = 99
+} KeyboardButton;
+
+typedef struct FileSelectState {
+    /* 0x00000 */ GameState state;
+    /* 0x000A4 */ Vtx* windowVtx;
+    /* 0x000A8 */ u8* staticSegment;
+    /* 0x000AC */ u8* parameterSegment;
+    /* 0x000B0 */ u8* titleSegment;
+    /* 0x000B8 */ View view;
+    /* 0x00220 */ SramContext sramCtx;
+    /* 0x00248 */ SkyboxContext skyboxCtx;
+    /* 0x00470 */ MessageContext msgCtx;
+    /* 0x12550 */ Font font;
+    /* 0x242E0 */ EnvironmentContext envCtx;
+    /* 0x243E0 */ Vtx* unk_243E0;
+    /* 0x243E4 */ Vtx* windowContentVtx;
+    /* 0x243E8 */ Vtx* keyboardVtx;
+    /* 0x243EC */ Vtx* nameEntryVtx;
+    /* 0x243F0 */ Vtx* keyboard2Vtx;
+    /* 0x243F4 */ u8 newf[4][6];
+    /* 0x2440C */ u16 sotCount[2];
+    /* 0x24410 */ u16 sotCount2[2];
```

Comment:
It's still not entirely clear why there's copies of each field. Everything not related to `sotCount` is only used in the copy function. `sotCount2` is used a little more but it's still not clear what it does

Reply:
Alright, can confirm that these doubled fields are suppose to be combined into 1 i.e. `u16 sotCount[4]`. The reason `sotCount2` is never accessed is because a variable is put into `sotCount` that can range from 0-3. Indices 0/1 being the default saves if you reset and indces 2/3 being the owl saves (when you do an owlsave for save file 1, you need both the 3-day reset saved and the owl temp saved). Worked it out by doing a deep-dive docs into the non-equivalent vtx function, and everything visually looks like it's now equivalent. More news soon :tm: